packages/installer/src/dappGet/fetch/DappGetFetcher.ts

```diff
@@ -78,7 +68,93 @@ export class DappGetFetcher {
       }
     }
-    // Case 3. unvalid semver version ("/ipfs/Qmre4..."): Asume it's the only version
+    // Case 3. invalid semver version ("/ipfs/Qmre4..."): Assume it's the only version
     return [versionRange];
   }
+
+  /**
+   * Merges optional dependencies into the main dependencies object if the corresponding 
+   * packages are installed.
+   *
+   * @param dependencies The main dependencies object to be merged into.
+   * @param optionalDependencies The optional dependencies to be checked and merged.
+   * @param installedPackages The list of currently installed packages.
+   */
+  private mergeOptionalDependencies(
+    dependencies: Dependencies,
+    optionalDependencies: Dependencies,
+    installedPackages: InstalledPackageData[]
+  ): void {
+    for (const [optionalDepName, optionalDepVersion] of Object.entries(optionalDependencies)) {
+      const isInstalled = installedPackages.some(
+        (installedPackage) => installedPackage.dnpName === optionalDepName
+      );
+
+      if (isInstalled) {
+        dependencies[optionalDepName] = optionalDepVersion;
+      }
+    }
+  }
+
+  /**
+   * Processes the dependencies by first filtering out those that are already satisfied by installed packages
+   * and then converting any remaining semver ranges to appropriate APM versions.
+   *
+   * @param dependencies The main dependencies object to be processed.
+   * @param installedPackages The list of currently installed packages.
+   */
+  private processDependencies(
+    dependencies: Dependencies,
+    installedPackages: InstalledPackageData[]
+  ): void {
+    for (const [depName, depVersion] of Object.entries(dependencies)) {
+      const installedPackage = installedPackages.find(
+        (pkg) => pkg.dnpName === depName
+      );
+
+      if (!validRange(depVersion))
+        throw new Error(`Invalid semver notation for dependency ${depName}: ${depVersion}`);
+
+      if (depVersion.includes('||') || depVersion.includes(' ')) {
+        throw new Error(`Unsupported version range for dependency ${depName}: ${depVersion}. Only simple ranges are supported`);
+      }
+
+      // Remove dependency if it is already satisfied by an installed package
+      if (installedPackage && satisfies(installedPackage.version, depVersion)) {
+        console.log(
+          `Dependency ${depName} is already installed with version ${installedPackage.version}`
+        );
+        delete dependencies[depName];
+        continue;
+      }
+
+      dependencies[depName] = this.parseSemverRangeToApmVersion(depVersion);
+    }
+  }
+
+  /**
+   * Parses a semver range and converts it to a version suitable for APM.
+   * Handles specific semver operators like ^, ~, >, >= and converts them to
+   * appropriate APM-compatible versions.
+   *
+   * @param semverRange The semver range to be parsed.
+   * @returns The parsed version as a string.
+   */
+  private parseSemverRangeToApmVersion(semverRange: string): string {
```

Comment:
For context: 

- The current package manifest version pattern is https://github.com/dappnode/DNP_DAPPMANAGER/blob/bafd376cf6b233da40e9d163477c02fa38adeec5/packages/schemas/src/schemas/manifest.schema.json#L42
```json
"pattern": "^((([0-9]+).([0-9]+).([0-9]+)))$",
```

- The repository SC only allows strict semver, see https://github.com/dappnode/DNP_DAPPMANAGER/blob/bafd376cf6b233da40e9d163477c02fa38adeec5/packages/toolkit/truffle/contracts/contracts_v0.4/repository.sol#L503
```solidity
    function isValidBump(uint16[3] _oldVersion, uint16[3] _newVersion) public pure returns (bool) {
        bool hasBumped;
        uint i = 0;
        while (i < 3) {
            if (hasBumped) {
                if (_newVersion[i] != 0) {
                    return false;
                }
            } else if (_newVersion[i] != _oldVersion[i]) {
                if (_oldVersion[i] > _newVersion[i] || _newVersion[i] - _oldVersion[i] != 1) {
                    return false;
                }
                hasBumped = true;
            }
            i++;
        }
        return hasBumped;
    }
```

Which means that we will kind of follow JS ecosystem where packages will have always a semver version and the package manager can parse the dependency target defined in the manifest (package.json)

Reply:
we should document this in the dev documentation letting devs know that they are allowed to do so

Reply:
Actually, I am able to generate and to install packages with semver ranges that are not considered in that pattern